test/extensions/filters/network/mysql_proxy/mysql_test_utils.cc

```diff
@@ -36,36 +38,72 @@ std::string MySQLTestUtils::encodeClientLogin(uint16_t client_cap, std::string u
   mysql_clogin_encode.setMaxPacket(MYSQL_MAX_PACKET);
   mysql_clogin_encode.setCharset(MYSQL_CHARSET);
   mysql_clogin_encode.setUsername(user);
-  std::string auth_resp(getAuthResp());
-  mysql_clogin_encode.setAuthResp(auth_resp);
-  std::string data = mysql_clogin_encode.encode();
-  std::string mysql_msg = BufferHelper::encodeHdr(data, seq);
-  return mysql_msg;
+  mysql_clogin_encode.setAuthResp(getAuthPluginData8());
+  Buffer::OwnedImpl buffer;
+  mysql_clogin_encode.encode(buffer);
+  BufferHelper::encodeHdr(buffer, seq);
+  return buffer.toString();
 }
 
 std::string MySQLTestUtils::encodeClientLoginResp(uint8_t srv_resp, uint8_t it, uint8_t seq_force) {
-  ClientLoginResponse mysql_loginok_encode{};
-  mysql_loginok_encode.setRespCode(srv_resp);
-  mysql_loginok_encode.setAffectedRows(MYSQL_SM_AFFECTED_ROWS);
-  mysql_loginok_encode.setLastInsertId(MYSQL_SM_LAST_ID);
-  mysql_loginok_encode.setServerStatus(MYSQL_SM_SERVER_OK);
-  mysql_loginok_encode.setWarnings(MYSQL_SM_SERVER_WARNINGS);
-  std::string data = mysql_loginok_encode.encode();
+  ClientLoginResponse mysql_login_resp_encode{};
+  mysql_login_resp_encode.initMessage(srv_resp);
+
+  switch (srv_resp) {
+  case MYSQL_RESP_OK:
+    mysql_login_resp_encode.asOkMessage().setAffectedRows(MYSQL_SM_AFFECTED_ROWS);
```

Comment:
As per comment above, when you restructure hierarchy, `asOKMessage` should not be needed. If you need it, please move it to test file as it is needed only for testing.